solutions/cses-1640.mdx

```diff
@@ -141,3 +123,135 @@ print("IMPOSSIBLE")
 </PySection>
 
 </LanguageSection>
+
+## Solution 2 - Two Pointers
+
+By keeping two pointers, one at each end of the list, we can greedily move the left and right pointers if the sum of the two elements are less than or greater than the target sum, finding the two indices in $\mathcal{O}(N)$ time.
+<LanguageSection>
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class TwoSum {
+	public static void main(String[] args) {
+		Kattio io = new Kattio();
+		long N, X;
+		N = io.nextInt();
+		X = io.nextInt();
+
+		List<Pair> a = new ArrayList<>();
+						  
+		// append the element and its index
+		for (int i = 0; i < N; i++) {
+			long x = io.nextInt();
+			a.add(new Pair(x, i + 1));
+		}
+		Collections.sort(a);
+		int i = 0, j = (int) N - 1;
+		while (i < j) {
+			// adjust the left and right pointers.
+			if (a.get(i).val + a.get(j).val > X)
+				j--;
+			else if (a.get(i).val + a.get(j).val < X)
+				i++;
+			else if (a.get(i).val + a.get(j).val == X) {
+				System.out.println(a.get(i).index + " " + a.get(j).index);
+				return;
+			}
+		}
+		io.println("IMPOSSIBLE");
+	}
+
+	static void sort(int[] a) {
```

Comment:
It doesn't seem like you're using this method

Reply:
you're right 👍